nixos/modules/services/web-apps/libretranslate.nix

```
@@ -0,0 +1,203 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.libretranslate;
+  ltmanageKeysCli = pkgs.writeShellScriptBin "ltmanage-keys" ''
+    set -a
+    export HOME="/var/lib/libretranslate"
+    sudo=exec
+    if [[ "$USER" != ${cfg.user} ]]; then
+      sudo='exec /run/wrappers/bin/sudo -u ${cfg.user} --preserve-env'
+    fi
+    $sudo ${cfg.package}/bin/ltmanage keys --api-keys-db-path ${cfg.dataDir}/db/api_keys.db "$@"
+  '';
+
+in {
+  options = {
+    services.libretranslate = {
+      enable = lib.mkEnableOption (lib.mdDoc "LibreTranslate service");
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.libretranslate;
+        defaultText = lib.literalExpression "pkgs.libretranslate";
+        description = lib.mdDoc "LibreTranslate package to use.";
+      };
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "libretranslate";
+        description = lib.mdDoc "User account under which libretranslate runs.";
+      };
+      group = lib.mkOption {
+        type = lib.types.str;
+        default = "libretranslate";
+        description = lib.mdDoc "Group account under which libretranslate runs.";
+      };
+      host = lib.mkOption {
+        description = lib.mdDoc "LibreTranslate host.";
+        type = lib.types.str;
+        default = "127.0.0.1";
+      };
+      port = lib.mkOption {
+        description = lib.mdDoc "LibreTranslate port.";
```

```suggestion
        description = lib.mdDoc "The address the application should listen on.";
        type = lib.types.str;
        default = "127.0.0.1";
      };
      port = lib.mkOption {
        description = lib.mdDoc "The the application should listen on.";
```
We shouldn't repeat the name of the service in each option.